src/Illuminate/Foundation/Auth/VerifiesEmails.php

```diff
@@ -35,6 +35,9 @@ public function verify(Request $request)
         if ($request->route('id') != $request->user()->getKey()) {
             throw new AuthorizationException;
         }
+        if ($request->route('hash') != hash('sha1', $request->user()->email)) {
```

Comment:
We have to use strict comparison to prevent false-positives:

```suggestion
        if ($request->route('hash') !== hash('sha1', $request->user()->email)) {
```

Reply:
Should actually use the timing safe comparison function, to avoid this hash being easily broken by timing analysis.